sky/spot/recovery_strategy.py

```diff
@@ -161,9 +161,16 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
                            detach_run=True,
                            _is_launched_by_spot_controller=True)
                 logger.info('Spot cluster launched.')
-            except exceptions.InvalidClusterNameError as e:
-                # The cluster name is too long.
-                raise exceptions.ResourcesUnavailableError(str(e)) from e
+            except exceptions.ResourcesUnavailableError as e:
+                if len(e.failover_reasons) == 1:
```

Comment:
A bit confused by:

- Why do we not need to catch InvalidClusterNameError anymore?
- Why do we do `if len(e.failover_reasons) == 1` in several places? What if then length is 0 or >1?